tests/x11regressions/remote_desktop/onetime_vncsession_xvnc_java.pm

```diff
@@ -0,0 +1,82 @@
+# Copyright (C) 2017 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Remote Login: One-time VNC Session with Jave applet and xvnc
+# Maintainer: Chingkai <qkzhu@suse.com>
+# Tags: tc#1586207
+
+use strict;
+use base 'basetest';
+use base 'x11regressiontest';
+use testapi;
+use lockapi;
+
+sub run() {
+    my $self = shift;
+
+    # Wait for supportserver if not yet ready
+    mutex_lock 'dhcp';
+    mutex_unlock 'dhcp';
+    mutex_lock 'xvnc';
+
+    # Make sure the client gets the IP address
+    x11_start_program 'xterm';
+    assert_screen 'xterm';
+    become_root;
+    assert_script_run 'dhclient';
+    type_string "exit\n";
+    send_key 'alt-f4';
+
+    # Start firefox
+    $self->start_firefox;
+    send_key 'esc';
+    send_key 'alt-d';
+    type_string "10.0.2.1:5801\n";
+    assert_screen 'firefox-ssl-untrusted';
+    send_key 'tab';
+    send_key 'ret';
+    send_key 'tab';
+    send_key 'ret';
+    assert_screen 'firefox-ssl-addexception', 60;
+    send_key 'alt-c';
+    wait_still_screen 3;
+    send_key 'alt-f10';
+    assert_and_click 'xvnc-firefox-activate-IcedTea';
+    assert_and_click 'xvnc-firefox-IcedTea-allow';
+    assert_screen 'xvnc-firefox-certification-warning';
+    assert_and_click 'xvnc-firefox-trust-ca';
+    assert_screen 'firefox-java-security';
+    assert_and_click 'firefox-java-securityrun';
+    assert_screen 'xvnc-firefox-verification-warning';
+    send_key 'ret';
+    if (check_screen 'xvnc-firefox-certification-warning2') {
+        send_key 'ret';
+    }
```

Comment:
IMHO you can replace by

```perl
assert_screen [qw(xvnc-firefox-certification-warning2 xvnc-firefox-dm)];
send_key 'ret' if (match_has_tag 'xvnc-firefox-certification-warning2');
```
and then continue with the next line to save waiting time.